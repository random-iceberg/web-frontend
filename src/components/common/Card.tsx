import React from "react";

/**
 * Props for the Card component.
 * @typedef {object} CardProps
 * @property {React.ReactNode} children - Content to be rendered inside the card.
 * @property {string} [className] - Additional CSS classes to apply to the card container.
 */
interface CardProps {
  children?: React.ReactNode; // Made children optional
  className?: string; // Allow passing additional classes
}

/**
 * A simple container component with padding, shadow, and rounded corners.
 * Useful for grouping content visually.
 *
 * @component
 * @param {CardProps} props - Props for the Card component.
 * @example
 * <Card className="mt-4 p-6">
 *   <h2>Card Title</h2>
 *   <p>Card content goes here.</p>
 * </Card>
 */
const Card: React.FC<CardProps> = ({ children, className = "" }) => {
  const baseStyles =
    "bg-white rounded-lg shadow p-4 hover:shadow-md transition-shadow"; // Default padding is p-4
  const combinedClassName = `${baseStyles} ${className}`.trim();

  return <div className={combinedClassName}>{children}</div>;
};

export default Card;
