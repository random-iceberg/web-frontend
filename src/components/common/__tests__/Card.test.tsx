import React from 'react';
import { render } from '@testing-library/react';
import Card from '../Card';

describe('Card Component', () => {
  it('renders correctly with children', () => {
    const { asFragment } = render(
      <Card>
        <p>This is card content.</p>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with children and className', () => {
    const { asFragment } = render(
      <Card className="my-custom-card extra-padding">
        <h2>Card Title</h2>
        <p>More card content.</p>
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no children', () => {
    const { asFragment } = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
  });
});
