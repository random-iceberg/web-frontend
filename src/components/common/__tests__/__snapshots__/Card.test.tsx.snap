// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly with children 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow p-4 hover:shadow-md transition-shadow"
  >
    <p>
      This is card content.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Card Component renders correctly with children and className 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow p-4 hover:shadow-md transition-shadow my-custom-card extra-padding"
  >
    <h2>
      Card Title
    </h2>
    <p>
      More card content.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Card Component renders correctly with no children 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-lg shadow p-4 hover:shadow-md transition-shadow"
  />
</DocumentFragment>
`;
